src/classes/ ffTestStud.cls

```diff
@@ -0,0 +1,194 @@
+                currentLimit = currentLimit - ListOfAdditionalMatches.size();
+                for(Account eachOrg : ListOfAdditionalMatches ) {   
+                    ListOfMatches.Add(eachOrg);
+                }
+            }
+         }    
+        //search # 5 match on city if no postcode
+        if ((!postCodeValid)&&(pCity!=null)&&(pCity!='') && (currentLimit>0)) {
+            if (pShowAllRecordTypes) {
+                    ListOfAdditionalMatches = [select id, Name, BillingStreet, BillingCity, BillingPostalCode, BillingState, Phone, Category__c, Owner.Alias, Record_Type_Name__c from Account where BillingCity=:pCity order by Name LIMIT :currentLimit ];
+                }
+                else {
+                    ListOfAdditionalMatches = [select id, Name, BillingStreet, BillingCity, BillingPostalCode, BillingState, Phone, Category__c, Owner.Alias, Record_Type_Name__c from Account where BillingCity=:pCity AND RecordTypeId =:pRecordTypeId order by Name LIMIT :currentLimit ];
+                    }
+                System.Debug ('##################Search5: ' + ListOfAdditionalMatches.size());
+                currentLimit = currentLimit - ListOfAdditionalMatches.size();
+                for(Account eachOrg : ListOfAdditionalMatches ) {   
+                    ListOfMatches.Add(eachOrg);
+                }
+            }
+        //search # 6 match on postcode only
+        if ((postCodeValid)&& (currentLimit>0)){
+            if (pShowAllRecordTypes) {
+                    ListOfAdditionalMatches = [select id, Name, BillingStreet, BillingCity, BillingPostalCode, BillingState, Phone, Category__c, Owner.Alias, Record_Type_Name__c from Account where BillingPostalCode=:pPostCode order by Name LIMIT :currentLimit ];
+                }
+                else {
+                    ListOfAdditionalMatches = [select id, Name, BillingStreet, BillingCity, BillingPostalCode, BillingState, Phone, Category__c, Owner.Alias, Record_Type_Name__c from Account where BillingPostalCode=:pPostCode AND RecordTypeId =:pRecordTypeId order by Name LIMIT :currentLimit ];
+                    }
+                System.Debug ('##################Search6: ' + ListOfAdditionalMatches.size());
+                currentLimit = currentLimit - ListOfAdditionalMatches.size();
+                for(Account eachOrg : ListOfAdditionalMatches ) {   
+                    ListOfMatches.Add(eachOrg);
+                }
+            }
+        
+        //return lstMatchingAccounts without duplicates and without the current record
+        Set <Account> conset = new Set<Account>();
+
+        for (Account s : ListOfMatches) {
+            if (conset.add(s)) {
+                if (s.id!=pCurrentRecord.id) {
+                    lstMatchingAccounts.add(s);
+                }
+            }
+        }
+        return lstMatchingAccounts;
+    }
+
+private Boolean IsValidPostCode(string pPostcode) {
```

Comment:
Methods should use the `lowerCamelCase` naming convention. I suggest to rename accordingly.